app/code/Magento/Catalog/Model/ProductRepository.php

```diff
@@ -280,6 +280,7 @@ public function __construct(
      */
     public function get($sku, $editMode = false, $storeId = null, $forceReload = false)
     {
+        $storeId = $storeId === null ? null : (int) $storeId;
```

Comment:
This change does not make sense. Numeric strings are also perfectly valid here.

Did you face with any concrete bug?

Reply:
For example:
vendor/magento/module-advanced-checkout/Controller/Cart/UpdateFailedItemOptions.php -> $this->_objectManager->get(\Magento\Store\Model\StoreManager::class)->getStore()->getId() this one returns string


$this->product = $this->productRepository->getById($entityId, true, $this->getStore()->getId()); -> this line here -> StagingLocator.php

etc

Reply:
Numeric strings are allowed, yes. 
But in the same process, if you call repo->getByIf(.., '1', ..) and repo->getByIf(.., 1, ..) you will load product twice.

Reply:
because cacheKey will be different

Reply:
Okej, nice catch, does actually make sense with such explanation :) Will review deeper a bit later, need to check this file for other occurrences.

Reply:
@sheepfy You opinion make sense. But I believe that `$storeId` type casting will be more safe on cache key calculation

```
$cacheKey = $this->getCacheKey([$editMode, $storeId === null ? $storeId : (int)$storeId]);
```

Reply:
@Den4ik  changes applied

Reply:
@sheepfy Thanks for update. I'll check it today

Reply:
@Den4ik you mean these?

```PHP Code Mess has found error(s):
/var/www/html/app/code/Magento/Catalog/Model/ProductRepository.php:42	The class ProductRepository has an overall complexity of 112 which is very high. The configured complexity threshold is 100.
/var/www/html/app/code/Magento/Catalog/Model/ProductRepository.php:526	The method save() has 115 lines of code. Current threshold is set to 100. Avoid really long methods.```